ReactiveXComponent/WebSocket/WebSocketMessageHelper.cs

```diff
@@ -135,13 +134,13 @@ public static WebSocketMessage DeserializeRequest(string request)
                 int firstDotIndex = firstPartBeforeJson.IndexOf('.');
 
                 // Assume it is an input on a custom type of topic if it doesn't have the format  topic = {key}.*.*.* (e.g. output.1_0.engine1.OrderService.CreationFacade)
-                key = firstDotIndex >= 0 ? firstPartBeforeJson.Substring(0, firstDotIndex) : WebSocketCommand.Input;
-                topic = tokensBeforeJson[0];
+                key = firstDotIndex > 0 ? firstPartBeforeJson.Substring(0, firstDotIndex) : WebSocketCommand.Input;

```

Comment:
this whole method is TOO complicated.

1. create a new method to extract the key:

`string GetKey(string s) - find a good name for s`

2. eventually, create a method that splits properly request intro its components;

`private string[] SplitRequest(string request)`

then the code will become much simpler;

```
var splitRequest = SplitRequest(request);

switch(splitRequest.length) {
case 1: json = splitRequest[0];
case 2: ...
default: error
} 
```